### DISCUSSION TITLE
blog/2023/08/15/matrix-multiply-logit-predict/

### DISCUSSION THREAD

**Discussion by giscus** [andrewheiss/ath-quarto#90] [Blog comments]
# blog/2023/08/15/matrix-multiply-logit-predict/

This is like basic stats stuff, but I can never remember how to do it—here’s how to use matrix multiplication to replicate the results of <code>predict()</code>

https://www.andrewheiss.com/blog/2023/08/15/matrix-multiply-logit-predict/

**Comment by kylebutts** [comment]
If you have a large dataset, sparse_model_matrix from the Matrix package can speed this up a bunch :-)

**Reply by andrewheiss** [reply]
Oooh I just tried this with my actual project data where we have a 576×6 matrix and it *was* faster!

```r
data_to_plug_in_cat_mat <- sparse.model.matrix(
  ~ 0 + ., data = data_to_plug_in_cat
)
```

**Reply by vincentarelbundock** [reply]
ha! came here to say this.

**Comment by conjugateprior** [comment]
> Since we don’t have an intercept term, we don’t need to create separate objects for the slopes and intercepts and can matrix multiply the new data matrix with the model coefficients

Happily, `model.matrix` will work just fine whether or not there's an intercept. It'll just add one if the formula wants one.

Further, fwiw, although it can seem like an arcane under-the-hood kind of R function, in teaching contexts I've found it's helpful to introduce `model.matrix` as soon and as regularly as possible. This seems to preempt a lot of regression `summary` interpretation questions, presumably because it leans into an important conceptualization / operationalization distinction, in which the data set holds the variables as you think of them, the model matrix holds the variables as the model thinks of them, and these are not necessarily the same.
